c++ training/ArraysVectors/Arrays-Vectors/ArraysVectors/StaticArray.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+template<typename T, int N>
+class TStaticArray
+{
+private:
+	T Data[N];
+
+public:
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	int GetSize() const
+	{
+		return N;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T& operator[](const int Index)
+	{
+		return Data[Index];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	const T& operator[](const int Index) const
+	{
+		return Data[Index];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T& Front()
+	{
+		return Data[0];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	const T& Front() const
+	{
+		return Data[0];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T& Back()
+	{
+		return Data[N - 1];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	const T& Back() const
+	{
+		return Data[N - 1];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T* GetData()
+	{
+		return &Data[0];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	const T* GetData() const
+	{
+		return &Data[0];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	bool IsEmpty() const
+	{
+		return N == 0;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	void Fill(const T& Copy)
+	{
+		for (T& Iteration : Data)
+		{
+			Iteration = T(Copy);
```

Comment:
> Why use a copy cosntructor here to pass into the assignment iterator. Why not use
> 
> ```
> Iteration = Copy
> ```
> 
> directly?

Shouldn't I define the = operator if I do it like that? In case our object behaves like strings or something